packages/curve-ui-kit/src/features/connect-wallet/lib/ConnectionContext.tsx

```diff
@@ -47,24 +51,43 @@ function useWagmiWallet() {
           account: { address }, // the ensName is set later when detected
           chainId: client.chain.id,
         }
+      hackSignerInCypress(wallet)
       return { wallet, provider: wallet ? new BrowserProvider(wallet.provider) : null }
     }, [address, client?.chain.id, client?.transport.request]) ?? null),
   }
 }
 
 /**
- * Hacks the signerAddress property of the library to match the wallet address in Cypress tests.
- * This is needed because the signer address in Cypress doesn't match the wallet's one when using a private key.
+ * Hacks the eth_accounts RPC method for Cypress tests to return the correct wallet address.
+ *
+ * **Problem**:
+ * - Wagmi with private key accounts (used in Cypress) doesn't expose accounts via eth_accounts
+ * - eth_accounts typically returns accounts managed by injected wallets (like MetaMask)
+ * - Private key accounts in wagmi/viem must be explicitly passed to actions, not hoisted to provider level
+ * - Our curve libraries use ethers BrowserProvider, which calls eth_accounts to determine signer address
+ * - This creates a mismatch where ethers can't find the correct signer address
+ *
+ * **Solution**:
+ * Override the provider's eth_accounts response to return the wallet address from the private key account.
+ * This bridges the incompatibility between wagmi's private key handling and ethers' expectation
+ * that accounts are available through the standard eth_accounts RPC method.
+ *
+ * @see https://wagmi.sh/vue/guides/viem#private-key-mnemonic-accounts
  */
-const hackSignerInCypress = (newLib: Libs[keyof Libs], wallet: Wallet | undefined) =>
-  isCypress &&
-  newLib?.signerAddress &&
-  wallet?.account.address &&
-  Object.defineProperty(newLib, 'signerAddress', {
-    get: () => wallet.account.address,
-    enumerable: true,
-    configurable: true,
-  })
+const hackSignerInCypress = (wallet: Wallet | undefined) => {
+  const provider = wallet?.provider
+
+  if (isCypress && provider) {
+    const originalRequest = provider.request.bind(provider)
+
+    provider.request = async (args: { method: string; [key: string]: any }) => {
+      if (args?.method === 'eth_accounts') {
```

Comment:
Ternary expression could be used

Reply:
https://github.com/curvefi/curve-frontend/commit/5ea560297d4dd36aa1952f3ba658296cbcd1a30f